### PR TITLE
apps/mlearning: tweak libcmsisdsp makefile

### PR DESCRIPTION
**Pull request by PeterBee97**
only one .c needed for each function group
add -flax-vector-conversions to avoid build error on gcc && M55

Signed-off-by: Peter Bee <bijunda1@xiaomi.com>

## Testing
Successfully compiled on Cortex-M55 platform

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@jerpelea could you take a look?

File: mlearning/cmsis/libcmsisnn/Makefile [hunk 1]
```diff
@@ -77,10 +77,6 @@ CSRCS += $(CMSIS5_NN)/ConvolutionFunctions/arm_convolve_HWC_q7_fast_nonsquare.c
 CSRCS += $(CMSIS5_NN)/ConvolutionFunctions/arm_depthwise_conv_s8.c
 CSRCS += $(CMSIS5_NN)/ConvolutionFunctions/arm_convolve_1x1_HWC_q7_fast_nonsquare.c
 CSRCS += $(CMSIS5_NN)/ConvolutionFunctions/arm_convolve_HWC_q7_basic_nonsquare.c
-CSRCS += $(CMSIS5_NN)/ConvolutionFunctions/arm_convolve_CHW_f32_basic_nonsquare.c
-CSRCS += $(CMSIS5_NN)/ConvolutionFunctions/arm_convolve_CHW_q7_basic_nonsquare.c
-CSRCS += $(CMSIS5_NN)/ConvolutionFunctions/arm_convolve_CHW_q15_basic_nonsquare.c
-CSRCS += $(CMSIS5_NN)/ConvolutionFunctions/arm_nn_CHW_mat_mult_kernel_q7_q15.c
```
Comment (xiaoxiang781216) [inline mlearning/cmsis/libcmsisnn/Makefile:83]: Please revert the change, these files are provided by:
https://github.com/apache/incubator-nuttx-apps/blob/master/mlearning/cmsis/cmsis-nn-support_nnabla.patch

Comment (PeterBee97) [reply mlearning/cmsis/libcmsisnn/Makefile]: Done

**Review by xiaoxiang781216 — APPROVED** [review]
